connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedConfig.java

```diff
@@ -265,7 +283,28 @@
                         (short) 3,
                         atLeast(1),
                         ConfigDef.Importance.LOW,
-                        STATUS_STORAGE_REPLICATION_FACTOR_CONFIG_DOC);
+                        STATUS_STORAGE_REPLICATION_FACTOR_CONFIG_DOC)
+                .define(CONNECT_PROTOCOL_CONFIG,
+                        ConfigDef.Type.STRING,
+                        CONNECT_PROTOCOL_DEFAULT,
+                        ConfigDef.LambdaValidator.with(
+                            (name, value) -> {
+                                try {
+                                    ConnectProtocolCompatibility.compatibility((String) value);
+                                } catch (Throwable t) {
+                                    throw new ConfigException(name, value, "Invalid Connect protocol "
```

Comment:
Would be nice to roll up this re-throw into ensureValid(), i.e. catch the exception and format a general error message there.

Reply:
Not a bad idea. But at the moment, I'd suggest taking improvements on this code in a separate PR